test-records.js

```diff
@@ -112,6 +113,56 @@ validator('companies', cdb_schema, (json) => {
                 msg: "Record has `quality` of `tested` but doesn't specify `required-elements`. See: https://github.com/datenanfragen/data/issues/811",
             });
     }
+    // whitespace check
+    Object.keys(json).forEach((key) => {
+        if (typeof json[key] === 'string' && json[key] !== json[key].trim())
+            errors.push(`Seems like \`${key}\` isn't trimmed, i.e. it contains leading or trailing whitespace.`);
+    });
+
+    //// ADDRESS FORMATTING
+    const address_lines = json['address'].split('\n');
+    if (address_lines.length < 2) errors.push('`address` is not formatted with newlines (\\n).');
+    for (line of address_lines) {
+        if (line !== line.trim()) {
+            ("`address` isn't trimmed, i.e. contains unnecessary whitespace.");
+        }
+    }
+    if (address_lines[0].trim() === json['name']) errors.push(' Record includes `name` in the `address`.');
+
+    // check if the last line is a country
+    // this test might produce false-positives, as it's a difficult thing to do
+    // we might consider more fuzzy matching if we get many reports of false positives
+    let last_line_is_country = false;
```

Comment:
Personally, I feel like this would be much cleaner and more readable if it were a `const llic = Object.entries(countries).some(...)` but oh well…